Implementacion de dimensiones y generar identidad

SantiagoHerrera1 opened the pull request.

Commit:
Implementacion de dimensiones y generar identidad

AndreyHorta reviewed (changes requested):
El pass debe ir al final de la funcion.  Ya que al inicio, antes del codigo, genera un error. 

DE la misma forma en generar identidad, sucede lo mismo con pass pero tambien tienes dos veces la definicion y genera error

SantiagoHerrera1 commented:
Hola Kevin, aunque solucione los errores que me comentas, el código no
funcionara ya que nuestro constructor es diferente, los códigos que yo
realice están hechos para funcionar con el constructor hecho en clase.
Respecto al Pass este no es necesario y no genera ningún problema con el
constructor que yo tengo, con la función generar identidad tenías razón la
escribí 2 veces. No se si te puedes comunicar con algún docente para
revisar lo del constructor ya que la idea es que todos lo tengamos igual y
así no genere ningún problema a ninguna de las partes.


CONSTRUCTOR E IMPRESOR

class Matriz:
    def __init__(self, cadena: str):
          if not (cadena[0] == "[" and cadena[-1] == "]"):
              print("No se puede construir la matriz.")
              return None

          cadena = cadena[1:-1]

          sfilas = cadena.split(";")
          self.nrows = len(sfilas)
          self.ncols = 0
          self.matriz = []

          for s in sfilas:
              filastr = s.split()
              filafloat = [float(e) for e in filastr]

              if self.ncols == 0:
                  self.ncols = len(filafloat)
              elif self.ncols != len(filafloat):
                  print("Número diferente de columnas en las filas.")
                  self.matriz = []
                  return None

              self.matriz.append(filafloat)

    def imprimir(self, decimales=2):
        for fila in self.matriz:
            for e in fila:
                print(f"{e:-8.2f}", end="")
            print(" ")

El mar, 19 nov 2024 a las 17:30, AndreyHorta ***@***.***>)
escribió:

> ***@***.**** requested changes on this pull request.
>
> El pass debe ir al final de la funcion. Ya que al inicio, antes del
> codigo, genera un error.
>
> DE la misma forma en generar identidad, sucede lo mismo con pass pero
> tambien tienes dos veces la definicion y genera error
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AndreyHorta/Matricial/pull/1#pullrequestreview-2446783614>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BLHIJLXKVFKCL3ONRMJW4UL2BO3ZHAVCNFSM6AAAAABSDERAN6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDINBWG44DGNRRGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
AVISO: Los servicios del correo ecci.edu.co <http://ecci.edu.co> son 
soportados tecnológicamente por © Google para toda la comunidad de la 
Universidad ECCI (Administrtivos, Docentes y Estudiantes), las opiniones 
que contenga este mensaje son exclusivas de su autor y no representan 
necesariamente la opinión de la Universidad ECCI. La Universidad ECCi no 
garantiza la utilización de un antivirus ya que se menciona los servicios 
son soportados © Google, si su antivirus detecta alguna anomalía por favor 
reportelo y elimine, la Universidad no se responsabiliza por los daños 
causados por cualquier virus transmitido en este correo electrónico.


La 
información contenida en este mensaje y en los archivos adjuntos es 
confidencial y reservada y está dirigida exclusivamente a su destinatario, 
sin la intención de que sea conocida por terceros, por lo tanto, de 
conformidad con las normas legales vigentes, su interceptación, 
sustracción, extravío, reproducción, lectura o uso esta prohibido a 
cualquier persona diferente. Se les exige expresamente a la comunidad ECCI 
(Administrtivos, Docentes y Estudiantes) que no realicen declaraciones 
difamatorias, no infrinjan ni autoricen ninguna infracción de las leyes de 
propiedad intelectual o cualquier otro derecho legal mediante 
comunicaciones por correo electrónico.La Universidad ECCI no garantizar 
guardar una copia de respaldo de los archivos, correos.


Si por error ha 
recibido este mensaje por favor discúlpenos, notifiquenoslo y elimínelo.

-- 
AVISO: Los servicios del correo ecci.edu.co <http://ecci.edu.co> son 
soportados tecnológicamente por © Google para toda la comunidad de la 
Universidad ECCI (Administrtivos, Docentes y Estudiantes), las opiniones 
que contenga este mensaje son exclusivas de su autor y no representan 
necesariamente la opinión de la Universidad ECCI. La Universidad ECCi no 
garantiza la utilización de un antivirus ya que se menciona los servicios 
son soportados © Google, si su antivirus detecta alguna anomalía por favor 
reportelo y elimine, la Universidad no se responsabiliza por los daños 
causados por cualquier virus transmitido en este correo electrónico.


La 
información contenida en este mensaje y en los archivos adjuntos es 
confidencial y reservada y está dirigida exclusivamente a su destinatario, 
sin la intención de que sea conocida por terceros, por lo tanto, de 
conformidad con las normas legales vigentes, su interceptación, 
sustracción, extravío, reproducción, lectura o uso esta prohibido a 
cualquier persona diferente. Se les exige expresamente a la comunidad ECCI 
(Administrtivos, Docentes y Estudiantes) que no realicen declaraciones 
difamatorias, no infrinjan ni autoricen ninguna infracción de las leyes de 
propiedad intelectual o cualquier otro derecho legal mediante 
comunicaciones por correo electrónico.La Universidad ECCI no garantizar 
guardar una copia de respaldo de los archivos, correos.


Si por error ha 
recibido este mensaje por favor discúlpenos, notifiquenoslo y elimínelo.

Commit:
Realice las respectivas correciones dichas por Kevin
